### PR TITLE
Registering documents on OS X

### PR DESCRIPTION
**Pull request by rgbkrk**
Thus begins registering nteract as a handler for `*.ipynb`. Haven't gotten this to actually do the registration, though I think these are the values. Since I didn't open XCode to help me make this, I wrote this snippet of node:

``` javascript
const plist = require('plist');

const fs = require('fs');

const data = {
  CFBundleDocumentTypes: [
    {
      CFBundleTypeExtensions: ['ipynb'],
      // CFBundleTypeIconFile ?
      CFBundleTypeName: 'io.nteract.notebook',
      CFBundleTypeRole: 'Editor',
      LSHandlerRank: 'Owner',
      LSItemContentTypes: ['io.nteract.notebook'],
    },
  ],
  UTExportedTypeDeclarations: [
    {
      UTTypeConformsTo: ['public.composite-content'],
      UTTypeDescription: 'nteract notebook',
      UTTypeIdentifier: 'io.nteract.notebook',
      UTTypeTagSpecification: {
        'public.filename-extension': ['ipynb'],
        'public.mime-type': 'application/x-ipynb+json',
      },
    },
  ],
};

fs.writeFileSync('./static/extend.plist', plist.build(data));
```

### CODE REVIEW

**Comment by rgbkrk** [comment]
Hold off on merging, I think I got the top level type on CFBundle wrong as well as the name itself...

File: static/extend.plist [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+          <array>
+            <string>ipynb</string>
+          </array>
+          <key>public.mime-type</key>
+          <string>application/x-ipynb+json</string>
+        </dict>
+      </dict>
+    </array>
+  </dict>
+</plist>
```
Comment (jdfreder) [inline static/extend.plist:47]: I'm surprised this is that big!

**Comment by rgbkrk** [comment]
Woo! This now works for accepting a file dragged and dropped on the icon either in the Dock or within Finder.

Still unsure how to make it work for spotlight. Maybe the plist has to be indexed?

File: static/extend.plist (hunk 1 above)
Comment (rgbkrk) [inline static/extend.plist:47]: How I eventually got this to work was to take a plist from an application I know well (Sketch), and apply it over here.

**Comment by rgbkrk** [comment]
Of note: the plist changes _only_ apply to the built app. You can't try this without building it and working with the built copy.

**Comment by rgbkrk** [comment]
This is all set now, and if we want to include the script that I use to generate this we certainly can. The JS is much easier to read (and much shorter) than the plist file.

**Comment by rgbkrk** [comment]
There's always the cheater mode from right click ipynb file -> get info:

screenshot 2016-04-19 00 00 16

screenshot 2016-04-18 23 59 18

File: static/extend.plist (hunk 1 above)
Comment (rgbkrk) [inline static/extend.plist:47]: On a side note, the JS for this is _way_ smaller:

``` js
const data = {
  CFBundleDocumentTypes: [
    {
      CFBundleTypeExtensions: ['ipynb'],
      // CFBundleTypeIconFile ?
      CFBundleTypeName: 'io.nteract.notebook',
      CFBundleTypeRole: 'Editor',
      LSHandlerRank: 'Owner',
      LSItemContentTypes: ['io.nteract.notebook'],
    },
  ],
  UTExportedTypeDeclarations: [
    {
      UTTypeConformsTo: ['public.composite-content'],
      UTTypeDescription: 'nteract notebook',
      UTTypeIdentifier: 'io.nteract.notebook',
      UTTypeTagSpecification: {
        'public.filename-extension': ['ipynb'],
        'public.mime-type': 'application/x-ipynb+json',
      },
    },
  ],
};
```

**Comment by jdfreder** [comment]
> Of note: the plist changes only apply to the built app. You can't try this without building it and working with the built copy.

This makes sense

> There's always the cheater mode from right click ipynb file -> get info:

So is the bottom screenshot a result of you setting it manually using finder?  Or the plist file being interpreted by OSX after a DND into Applications/ ?

**Comment by rgbkrk** [comment]
The bottom screenshot is me setting it manually since I couldn't open it from Finder. Drag and drop to nteract app icon work fine with this.

¯\_(ツ)_/¯

**Comment by minrk** [comment]
This is awesome.

**Pull request merged**